gorb/gorb_test.go

```diff
@@ -0,0 +1,149 @@
+		}
+	}
+
+	if h.responsePrimer[h.requestCounter].statusCode != 0 {
+		w.WriteHeader(h.responsePrimer[h.requestCounter].statusCode)
+	}
+	if h.responsePrimer[h.requestCounter].response != "" {
+		data, _ := json.Marshal(h.responsePrimer[h.requestCounter].response)
+		w.Write(data)
+	}
+
+}
+
+var _ = Describe("Gorb", func() {
+	var (
+		gorb             controller.Updater
+		server           *httptest.Server
+		serverURL        string
+		responsePrimer   []gorbResponsePrimer
+		recordedRequests []gorbRecordedRequest
+		gorbH            *gorbHandler
+	)
+
+	BeforeSuite(func() {
+		responsePrimer = []gorbResponsePrimer{}
+		recordedRequests = []gorbRecordedRequest{}
+		gorbH = &gorbHandler{responsePrimer: responsePrimer, recordedRequest: recordedRequests}
+		server = httptest.NewServer(gorbH)
+
+		serverURL = server.URL
+		log.Info("url ", serverURL)
+		gorb, _ = New(serverURL, instanceIP, drainImmediately, servicesName, servicesPort, backendWeight, backendMethod, vipLoadbalancer)
+	})
+
+	BeforeEach(func() {
+		gorbH.responsePrimer = []gorbResponsePrimer{}
+		gorbH.recordedRequest = []gorbRecordedRequest{}
+		gorbH.requestCounter = 0
+	})
+
+	AfterSuite(func() {
+		server.Close()
+	})
+
+	Describe("Health endpoint", func() {
+		It("should be healthy when status code is 200", func() {
+			gorbH.responsePrimer = append(gorbH.responsePrimer, gorbResponsePrimer{statusCode: 200})
+			err := gorb.Health()
+			Expect(err).NotTo(HaveOccurred())
+		})
+		It("should return error when status code is not 200", func() {
+			gorbH.responsePrimer = append(gorbH.responsePrimer, gorbResponsePrimer{statusCode: 500})
+			err := gorb.Health()
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Describe("Update backends", func() {
+		It("should add itself as new backend", func() {
+			gorbH.responsePrimer = append(gorbH.responsePrimer, gorbResponsePrimer{statusCode: 404})
+			gorbH.responsePrimer = append(gorbH.responsePrimer, gorbResponsePrimer{statusCode: 200})
+			err := gorb.Update(controller.IngressEntries{})
+			Expect(len(gorbH.recordedRequest)).To(Equal(2))
+			Expect(gorbH.recordedRequest[1].url.RequestURI()).To(Equal(fmt.Sprintf("/service/http-proxy/node-%s", instanceIP)))
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should modify when already exist", func() {
+			gorbH.responsePrimer = append(gorbH.responsePrimer, gorbResponsePrimer{statusCode: 200})
+			gorbH.responsePrimer = append(gorbH.responsePrimer, gorbResponsePrimer{statusCode: 200})
+			err := gorb.Update(controller.IngressEntries{})
+			Expect(len(gorbH.recordedRequest)).To(Equal(2))
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should remove itself on shutdown", func() {
+			gorbH.responsePrimer = append(gorbH.responsePrimer, gorbResponsePrimer{statusCode: 200})
+			gorbH.responsePrimer = append(gorbH.responsePrimer, gorbResponsePrimer{statusCode: 200})
+			err := gorb.Stop()
+			Expect(len(gorbH.recordedRequest)).To(Equal(2))
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+	})
+})
```

Comment:
I don't particularly mind here since it's not like we're mixing both testing approachs in a single package. But we should probably ask the other people on the team what they think.